### PR TITLE
Add GitHub actions workflow for CI tests

### PR DESCRIPTION
**Pull request by stomar**
Use GitHub actions workflow for running `rake ci`.

### CODE REVIEW

**Comment by stomar** [comment]
**Note: WIP**

My first trial with GitHub actions.

It seems to work, but compared to Travis takes forever (installation of the gem bundle takes about 4 to 5 minutes),

**Comment by stomar** [comment]
I added caching for the gem bundle, now took 1 min 20 s. (BTW, Travis after 5 min still pending...)

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        ruby: [ '2.6.3' ]
```
Comment (JuanitoFatas) [inline .github/workflows/ci.yml]: 2.6.5?

https://github.com/ruby/www.ruby-lang.org/blob/cfcb63e3264df8dba8dec7138b09e663ef540f40/.travis.yml#L3

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        with:
+          ruby-version: ${{ matrix.ruby }}
+          architecture: 'x64'
+      - name: Cache gem bundle
+        uses: actions/cache@v1
+        with:
+          path: vendor/bundle
+          key: bundle-${{ hashFiles('**/Gemfile.lock') }}
+      - name: Install bundler
+        run: gem install bundler
```
Comment (JuanitoFatas) [inline .github/workflows/ci.yml]: would `--no-document` help here?

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        ruby: [ '2.6.3' ]
+      fail-fast: true
+    runs-on: ubuntu-latest
+    env:
+      CUSTOM_RUBY_VERSION: ${{ matrix.ruby }}
```
Comment (JuanitoFatas) [inline .github/workflows/ci.yml]: What's this for? 🙋‍♂

File: .github/workflows/ci.yml [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+    name: Ruby ${{ matrix.ruby }}
+    steps:
+      - name: Dump environment
+        run: env | sort
+      - name: Checkout ruby/www.ruby-lang.org
+        uses: actions/checkout@v1
+        with:
+          fetch-depth: 1
```
Comment (JuanitoFatas) [inline .github/workflows/ci.yml]: Awesome 🥇

File: .github/workflows/ci.yml (hunk 1 above)
Comment (stomar) [reply .github/workflows/ci.yml]: See above; 2.6.5 currently is not provided by the setup-ruby action.

Comment (JuanitoFatas) [reply .github/workflows/ci.yml]: 🤦‍♂  I see. Sorry.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (JuanitoFatas) [reply .github/workflows/ci.yml]: To work around GitHub Actions not supported 2.6.5.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (stomar) [reply .github/workflows/ci.yml]: Not much :smile:, but a good suggestion nevertheless.

File: .github/workflows/ci.yml (hunk 3 above)
Comment (stomar) [reply .github/workflows/ci.yml]: To overide the default Ruby version in the Gemfile (2.6.5); else it would abort.

**Comment by stomar** [comment]
I do not know yet why I have to use `hashFiles('**/Gemfile.lock')` instead of simply `hashFiles('Gemfile.lock')` (which does not work). I'm still investigating.

**Comment by stomar** [comment]
I'm confident enough about this to merge and test under real conditions.

**Comment by hsbt** [comment]
@stomar Thanks.

**Pull request merged**